### DISCUSSION TITLE
Csla and [Transactional(TransactionalTypes.TransactionScope)]

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3407] [Questions]
We have a Csla 4.XX project and we thought of using [Transactional(TransactionalTypes.TransactionScope)] for updates and inserts.
But when FieldManager.UpdateChildren(this); is called we are getting
'INSERT statement conflicted with FOREIGN key constraint' error in child object data access layer.

I debugged the code and the primarykey from parent table ( also a property in root object ) is passed correctly to the child object's DAL.
We are using stored procs and ADO.NET.

Kind Regards

**Comment by rockfordlhotka** [comment]
And your code works without the transactional wrapper?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  yes, it works without the transactional wrapper.

**Reply by rockfordlhotka** [reply]
That's strange, and I have no idea why this would be happening. Something about being in a transaction must be changing how the database is handling the key relationship or enforcement.
